aequilibrae/project/network/osm/osm_downloader.py

```diff
@@ -23,16 +23,19 @@
 from aequilibrae.context import get_logger
 from aequilibrae.parameters import Parameters
 from aequilibrae.utils.signal import SIGNAL
+from aequilibrae.utils.interface.worker_thread import WorkerThread
 from .osm_params import http_headers, memory
 
 
-class OSMDownloader:
-    downloading = SIGNAL(object)
+class OSMDownloader(WorkerThread):
+    signal = SIGNAL(object)
 
     def __emit_all(self, *args):
-        self.downloading.emit(*args)
+        self.signal.emit(*args)
```

Comment:
If ___emit_all_ only does this, then we can definitely remove it and just emit directly from all lines